### PR TITLE
Remove Bip44Address, AddressWithUnspentOutputs::{key_index, internal}

### LINKED ISSUE #2025: Remove `AddressWithUnspentOutputs::{key_index, internal}`

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Remove Bip44Address, AddressWithUnspentOutputs::{key_index, internal} as they are no longer needed
Also made `AddressWithUnspentOutputs` private since we don't need it exposed

## Links to any relevant issues

Closes https://github.com/iotaledger/iota-sdk/issues/2025

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

`cargo test`

### CODE REVIEW

**Review by Alex6323 — APPROVED** [review]

File: sdk/src/types/block/protocol/samples.rs [hunk 1]
```diff
@@ -242,7 +242,7 @@ impl ProtocolParameters {
         self.rewards_parameters.final_target_rewards_rate = (self.token_supply()
             * self.rewards_parameters().reward_to_generation_ratio() as u64
             * self.mana_parameters().generation_rate() as u64)
-            >> self.mana_parameters().generation_rate_exponent() - self.slots_per_epoch_exponent();
+            >> (self.mana_parameters().generation_rate_exponent() - self.slots_per_epoch_exponent());
```
Comment (thibault-martinez) [inline sdk/src/types/block/protocol/samples.rs:245]: We're 100% sure about that? Because it actually does change the computation

Comment (Thoralf-M) [reply sdk/src/types/block/protocol/samples.rs:245]: It doesn't change https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cd273bf1d04773378b475cd7c2d47d6e

Comment (Thoralf-M) [reply sdk/src/types/block/protocol/samples.rs:245]: https://doc.rust-lang.org/reference/expressions.html#expression-precedence `-` has precedence over `>>`

Comment (thibault-martinez) [reply sdk/src/types/block/protocol/samples.rs:245]: Yeah I just read this page, interesting, I would have expected the opposite, thanks

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**